maas/resource_maas_vm_host.go

```diff
@@ -256,20 +254,27 @@ func resourceVMHostDelete(ctx context.Context, d *schema.ResourceData, meta inte
 		return diag.FromErr(err)
 	}
 
-	// If the VM host was deployed from a machine, release the machine.
-	if vmHost.Host.SystemID != "" {
-		// Release machine
-		err = client.Machines.Release([]string{vmHost.Host.SystemID}, "Released by Terraform")
-		if err != nil {
-			return diag.FromErr(err)
-		}
-		// Wait machine to be released
-		_, err = waitForMachineStatus(ctx, client, vmHost.Host.SystemID, []string{"Releasing"}, []string{"Ready"})
-		if err != nil {
-			return diag.FromErr(err)
+	// Check if VM host was linked to a dynamic machine and if yes, return
+	// Dynamic machines are deleted by MAAS when their VM hosts are deleted.
+	// This information is not directly available from the API.
+	_, err = client.Machine.Get(vmHost.Host.SystemID)
```

Comment:
I don't understand what you are trying to achieve here.

A VM host is a machine designed to run virtual machines.
`vmHost.Host.SystemID` will contain `systemID` of a physical machine that was deployed with "Register as MAAS KVM host" option, right?

My understanding is that we want to delete all _linked_ VMs that are linked to a VM host. Or am I wrong?

Reply:
VM Host can also be an existing VM Host aka a machine not created by MAAS but is running LXD or Virsh and we register its credentials so MAAS can register it as a VM Host. To achieve that, MAAS is using the concept of dynamic machine. For example, try to add your laptop as a VM Host to your lab MAAS. Obviously the laptop is not a machine owned by MAAS, but you can see it being displayed as a maas machine on machines tab. This is a dynamic machine and when VM Host is deleted, that abstraction of a machine is also deleted by MAAS with specific logic. In such cases, Terraform will get an error because it will try to ask MAAS to release your laptop, since `systemID` is pointing to something that does not exist. #137 has some additional info.

Reply:
This is not related with linked VMs of a VM Host at all. Instead it is taking care of the machine that acts as a VM host which can be either MAAS owned or something external represented as dynamic machine